src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java

```diff
@@ -213,6 +222,14 @@ private ConnectionState doReadHTTPInitial(HttpRequest httpRequest) {
         // Send the request through the clientToProxyRequest filter, and respond with the short-circuit response if required
         HttpResponse clientToProxyFilterResponse = currentFilters.clientToProxyRequest(httpRequest);
 
+        // Requesting the proxy directly must be answered elsewhere it causes an
+        // endless loop
+        //
+        if (clientToProxyFilterResponse == null && isProxyRequested()) {
+            clientToProxyFilterResponse = new DefaultFullHttpResponse(HttpVersion.HTTP_1_1,
```

Comment:
I'm wondering if we should treat this as a special case of clientToProxyFilterResponse. Instead, perhaps we should follow the same pattern we do with Bad Gateways, where the check for proxy self-requests would happen below the `if (clientToProxyFilterResponse != null)` block ends at line 242.